datetime_tz/pytz_abbr.py

```diff
@@ -47,19 +47,63 @@
 import pytz
 import pytz.tzfile
 
+try:
+  basestring
+except NameError:
+  # pylint: disable=redefined-builtin
+  basestring = str
+
 
 class tzabbr(datetime.tzinfo):
   """A timezone abbreviation.
 
-  *WARNING*: This is not a tzinfo implementation! Trying to use this as tzinfo
-  object will result in failure.  We inherit from datetime.tzinfo so we can get
-  through the dateutil checks.
+  *WARNING*: This is a proxy object for an underlying `pytz` object only
+  intended to allow dateutil to probe the localized datetime for information.
+  datetime_tz's parsing logic will replace it with the relevant pytz zone
+  before returning to the end user.
   """
-  pass
+
+  def __init__(self, abbr, name, region, zone, dst):
+    super(tzabbr, self).__init__()
+
+    self.abbr = abbr
+    self.name = name
+    self.region = region
+
+    if isinstance(zone, basestring):
+      zone = pytz.timezone(zone)
+
+    self.zone = zone
+    self.dst = dst
+
+  def _get_localized(self, dt):
+    # To make this a fully-functioning pass-through to the underlying pytz
+    # zone, we would want to use `fold` to set `is_dst`, but since this is
+    # only a temporary proxy for the zone, we will fix the DST status
+    return self.zone.localize(dt.replace(tzinfo=None), is_dst=self.dst)
+
+  def tzname(self, dt):
+    return self._get_localized(dt).tzname()
+
+  def utcoffset(self, dt):
+    return self._get_localized(dt).utcoffset() # pragma: no cover
+
+  def dst(self, dt):
```

Comment:
dst needs to be a method (as this is) but above in the constructor you set `.dst` as an attribute.

Reply:
Oooh, good catch, changed it to `is_dst`.

Would be nice to get actual coverage of this and `utcoffset`, but it seems wrong to add a test for something that should not really be part of the public interface.